instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/semconv/http/HttpClientAttributesExtractor.java

```diff
@@ -141,8 +151,69 @@ private static String stripSensitiveData(@Nullable String url) {
     }
 
     if (atIndex == -1 || atIndex == len - 1) {
-      return url;
+      return redactSensitiveParameters ? redactUrlParameters(url) : url;
+    }
+
+    String afterUserPwdRedaction = url.substring(atIndex);
+    return url.substring(0, schemeEndIndex + 3)
+        + "REDACTED:REDACTED"
+        + (redactSensitiveParameters
+            ? redactUrlParameters(afterUserPwdRedaction)
+            : afterUserPwdRedaction);
+  }
+
+  private static String redactUrlParameters(String urlpart) {
```

Comment:
In my opinion, the logic is a bit complex, in order to improve maintainability, how about adding comments that describe how to redact the URL or related reference above the method?

Reply:
I have added comments